core/domain/translation_domain.py

```diff
@@ -18,9 +18,262 @@
+                necessary logic to get all translatable fields in a
+                translatable object.
+        """
+        raise NotImplementedError('Must be implemented in subclasses.')
+
+    def get_all_contents_which_need_translations(
+        self,
+        entity_translation: EntityTranslation
+    ) -> List[TranslatableContent]:
+        """Returns a list of TranslatableContent instances which need new or
+        updated translations.
+
+        Args:
+            entity_translation: EntityTranslation. An object storing the
+                existing translations of an entity.
+
+        Returns:
+            list(TranslatableContent). Returns a list of TranslatableContent.
+        """
+        contents_which_need_translation = []
+        content_ids_for_translated_contents = (
+            entity_translation.translations.keys())
+        translatable_content_list = (
+            self.get_translatable_contents_collection()
+            .translatable_contents.values())
+
+        for translatable_content in translatable_content_list:
+            if translatable_content.content == '':
+                continue
+
+            if (
+                translatable_content.content_id not in
+                content_ids_for_translated_contents
+            ):
+                contents_which_need_translation.append(translatable_content)
+            elif (
+                entity_translation.translations[
+                translatable_content.content_id].needs_update
+            ):
+                contents_which_need_translation.append(translatable_content)
+
+        return contents_which_need_translation
+
+
+class EntityTranslation:
+    """The EntityTranslation represents an EntityTranslationsModel for a given
+    version of an entity in a given language.
+
+    Args:
+        entity_id: str. The id of the corresponding entity.
+        entity_type: TranslatableEntityType. The type
+            of the corresponding entity.
+        entity_version: str. The version of the corresponding entity.
+        language_code: str. The language code for the corresponding entity.
+        translations: dict(str, TranslatedContent). The translated content.
+    """
+
+    def __init__(
+        self,
+        entity_id: str,
+        entity_type: feconf.TranslatableEntityType,
+        entity_version: int,
+        language_code: str,
+        translations: Dict[str, TranslatedContent]
+    ):
+        self.entity_id = entity_id
+        self.entity_type = entity_type.value
+        self.entity_version = entity_version
+        self.language_code = language_code
+        self.translations = translations
+
+
+class TranslatableContent:
+    """TranslatableContent represents a content of a translatable object which
+    can be translated into multiple languages.
+
+    Args:
+        content_id: str. The id of the corresponding content.
+        content: ContentInTranslatableContent. The content which can be
```

Comment:
You can't both say this and also have a docstring above that's something like "The id of the corresponding translatable content." -- does "content" include the ID or does it not?

Perhaps use the suggestion above re calling things like this the content_value instead, and leaving "content" for the composite object.